docs/source/features/long_context_recipe.rst

```diff
@@ -0,0 +1,117 @@
+.. _long_context_recipe:
+
+Long Context Recipe
+------------------------
+
+Long Context Model Training enhances the capability of Large Language Model to handle long context inputs. Longer sequence length is beneficial for many NLP tasks, such as document-level summarization, long document classification, and long document question answering. NeMo provides a recipe to train long context models like Llama-3, Mixtral and Nemotron.
+
+
+Access Long Context Recipe
+========================
+
+NeMo 2.0 is providing a tested recipe to train long context models. The recipe is available in the `NeMo/nemo/collections/llm/gpt/trainer` directory.
+
+Currently we are supporting the following models:
+- Llama-3
+- Mixtral
+- Nemotron
+
+Here are charts that shows the different sequence lengths that are supported by each model with different sizes:
+
+
+Llama-3
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. list-table:: Llama-3
+   :widths: 30 25 25
+   :header-rows: 1
+
+   * - Sequence Length
+     - 8B
+     - 70B
+   * - 16k
+     - Yes
+     - Yes
+   * - 64k
+     - Yes
+     - Yes
+   * - 128k
+     - TODO
+     - TODO
+
+Mixtal
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. list-table:: Mixtral
+   :widths: 30 25 25 25
+   :header-rows: 1
+
+   * - Sequence Length
+     - 8x3B
+     - 8x7B
+     - 8x22B
+   * - 16k
+     - Yes
+     - Yes
+     - TODO
+   * - 64k
+     - Yes
+     - Yes
+     - TODO
+   * - 128k
+     - TODO
+     - TODO
+     - TODO
+
+Nemotron (Not yet supported in NeMo 2.0)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. list-table:: Nemotron
+   :widths: 30 25 25
+   :header-rows: 1
+
+   * - Sequence Length
+     - 15B
+     - 340B
+   * - 16k
+     - TODO
+     - TODO
+   * - 64k
+     - TODO
+     - TODO
+   * - 128k
+     - TODO
+     - TODO
+
+
+Context Parallelism
+========================
+
+Context Parallelism (CP) is a method for parallelizing the processing of neural network activations across multiple GPUs, partitioning the input tensors in the sequence dimension.
+Unlike Sequence Parallelism (SP) that partitions the activations of specific layers, CP divides the activations of all layers.
+
+CP is critical for training long context models, as it allows the model to handle longer sequences by distributing the sequence activations across multiple GPUs. This method reduces the memory footprint and computational cost of processing long sequences.
+
+Enable Context Parallelism
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To activate CP in the NeMo framework, set the ``context_parallel_size`` parameter in the model configuration. This parameter specifies the number of GPUs among which the model's sequence activations are distributed.
```

Comment:
revise.

To activate CP in the NeMo Framework, set the ``context_parallel_size`` parameter in the model configuration. This parameter specifies the number of GPUs across which the model’s sequence activations are distributed.